packages/adp-tooling/src/preview/client/dialogs/fragment.ts

```diff
@@ -0,0 +1,51 @@
+/** sap.ui.core */
+import XMLView from 'sap/ui/core/mvc/XMLView';
+import type UI5Element from 'sap/ui/core/Element';
+import Controller from 'sap/ui/core/mvc/Controller';
+
+/** sap.ui.rta */
+import type RuntimeAuthoring from 'sap/ui/rta/RuntimeAuthoring';
+
+/** sap.ui.dt */
+import type ContextMenu from 'sap/ui/dt/plugin/ContextMenu';
+
+import type AddFragment from '../controllers/AddFragment.controller';
+import type { BaseDialog } from './base';
+
+/**
+ * Handles creation of the dialog, fills it with data
+ */
+export default class FragmentDialog implements BaseDialog {
```

Comment:
Seems like a bit of an overhead to have a UI5 class here since it only offers one function that is called exactly once.
Do you expect anything else here in the future that would require a class/instance keeping a state? Otherwise, I would recommend to just export the init function (with an additional parameter `rta`)